### PR TITLE
Application: add an option to start with the settings visible

### PR DESCRIPTION
**Pull request by ogoffart**
Relates to #7018

### CODE REVIEW

File: docs/modules/ROOT/pages/advanced_usage/options.adoc [hunk 1]
```diff
@@ -10,6 +10,7 @@ The other options are:
 [options="header"]
 |===
 | Option | Description
+| `--showsettings` | Open the settings dialog while starting.
```
Comment (mmattel) [inline docs/modules/ROOT/pages/advanced_usage/options.adoc]: Why do you only show the long form in help but dont show the short form `-s` ?
There are many examples in eg. occ commands in docs where we show both, just in the way it has been designed.

Comment (ogoffart) [reply docs/modules/ROOT/pages/advanced_usage/options.adoc]: Yes you are right, i should add the `-s`  in the docs

File: docs/modules/ROOT/pages/advanced_usage/options.adoc [hunk 2]
```diff
@@ -10,6 +10,7 @@ The other options are:
 [options="header"]
 |===
 | Option | Description
+| `--showsettings` or `-s` | Open the settings dialog while starting.
```
Comment (mmattel) [inline docs/modules/ROOT/pages/advanced_usage/options.adoc]: Only a detail: we never use `or` when having a short and long option.
We would write:
|  `-s` +
`--showsettings` | Open the settings dialog while starting

See https://doc.owncloud.com/server/10.3/admin_manual/configuration/server/occ_command.html#background-jobs-selector
Executing a background job

Comment (mmattel) [reply docs/modules/ROOT/pages/advanced_usage/options.adoc]: I know you will not like it, but you have to make a new line after the plus symbol, exactly as written. This is due to the way how ascidoc works. The `+` sign at the end of a line shows, that the next line is part of that block but a new line (like you have in MS Word shift-return making a new line in the same paragraph while a return makes a new line with a new paragraph). Having that, the table renders correctly.

Comment (ogoffart) [reply docs/modules/ROOT/pages/advanced_usage/options.adoc]: Fixed, thanks for explaining me. (I thought the + was meant as a literal +)

Comment (mmattel) [reply docs/modules/ROOT/pages/advanced_usage/options.adoc]: Approved from the docs pov 😄 
Marking as resolved

**Comment by TheOneRing** [comment]
Looks good but I'm  not 100% sure about the use case.
Should the desktop entry (shortcut/desktop file) pass -s?
I mean especially on gnome where we no longer have a system tray the settings window should show when we start from the desktop file.

Btw: should we create a low lvl ticket to port to QCommandLineParser

**Comment by ogoffart** [comment]
> Should the desktop entry (shortcut/desktop file) pass -s?

Yes, i'd say so.

> Btw: should we create a low lvl ticket to port to QCommandLineParser

Not sure how good QCommandLineParser is when we pass the option accross process if the application is already running.

**Comment by TheOneRing** [comment]
> > Should the desktop entry (shortcut/desktop file) pass -s?
> 
> Yes, i'd say so.
> 
> > Btw: should we create a low lvl ticket to port to QCommandLineParser
> 
> Not sure how good QCommandLineParser is when we pass the option accross process if the application is already running.
Should be possible https://doc.qt.io/qt-5/qcommandlineparser.html#parse

**Pull request merged**